packages/client/src/app/personal_page/stores/personal_page.store.js

```diff
@@ -0,0 +1,111 @@
+import { defineStore, storeToRefs } from 'pinia';
+import { ref } from 'vue';
+import { useAuthStore } from '../../auth/index.js';
+import { GroupService } from '../../group/services/group.service.js';
+import { cardResource } from '../resources/card.resource.js';
+import { BadgeService } from '../services/index.js';
+
+export const usePersonalPageStore = defineStore('personal_page', () => {
+  /** @type {Array} */
+  let earnedBadges = ref({});
+
+  /** @type {Array} */
+  let groups = ref({});
+
+  /** @type {import('vue').Ref<boolean>} */
+  const loadingBadges = ref(true);
+
+  /** @type {import('vue').Ref<boolean>} */
+  const loadingGroups = ref(true);
+
+  /**
+   * @param userId {number}
+   * @return {void}
+   */
+  async function getEarnedBadges(userId) {
+    try {
+      await BadgeService.instance()
+        .getBadges(userId)
+        .then((data) => {
+          earnedBadges.value = data;
+        });
+    } catch (error) {
+      console.error(error);
+      earnedBadges.value = null;
+    } finally {
+      loadingBadges.value = false;
+    }
+  }
+
+  /**
+   * @param userId {number}
+   * @return {void}
+   */
+  async function getGroups(userId) {
+    try {
+      await GroupService.instance()
+        .getAllForUser(userId)
+        .then((data) => {
+          groups.value = cardResource(data);
+        });
+    } catch (error) {
+      console.error(error);
+      earnedBadges.value = null;
+    } finally {
+      loadingGroups.value = false;
+    }
+  }
+
+  /**
+   * @param name {String}
+   * @return {Object}
+   */
+  async function createGroup(name) {
+    const { user } = storeToRefs(useAuthStore());
+    let group = null;
+
+    try {
+      await GroupService.instance()
```

Comment:
Instead of the `.then` style you can make fully use of the `await` structure by structuring your code like this:
```suggestion
      group = await GroupService.instance().createForUser(user.value.id, name);
      
      loadingGroups.value = true;
      getGroups(user.value.id);
```